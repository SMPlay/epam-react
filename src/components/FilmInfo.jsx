import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { createUseStyles } from "react-jss";
import { Button } from "./Button";
import { Logo } from "./Logo";

const useStyles = createUseStyles({
  currentFilm: {},
  navigation: {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
  },
  aboutFilm: {
    display: "flex",
    justifyContent: "space-between",
    marginTop: "10px",
  },
  aboutFilmImg: {
    height: "500px",
  },
  aboutFilmText: {
    maxWidth: "740px",
  },
  aboutFilmTextTitle: {
    color: "rgb(255, 41, 41)",
    fontSize: "30px",
  },
  aboutFilmTextTagline: {
    fontSize: "18px",
    fontWeight: "400px",
    color: "white",
    marginTop: "8px",
  },
  time: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between",
    width: "150px",
    marginTop: "12px",
  },
  timeItem: {
    display: "block",
    color: "white",
  },
  overview: {
    marginTop: "35px",
    color: "white",
  },
});

export const FilmInfo = ({ filmInfo }) => {
  const {
    title,
    poster_path,
    tagline,
    release_date,
    runtime,
    overview,
  } = filmInfo;

  const classes = useStyles();

  return (
    <div className={classes.currentFilm}>
      <div className={classes.navigation}>
        <Logo />
        <Link to="/">
          <Button
            width="100px"
            type="primary"
            background="white"
            color="rgb(255, 41, 41)"
          >
            search
          </Button>
        </Link>
      </div>
      <div className={classes.aboutFilm}>
        <img
          className={classes.aboutFilmImg}
          src={poster_path}
          alt={title + tagline}
        />
        <div className={classes.aboutFilmText}>
          <h2 className={classes.aboutFilmTextTitle}>{title}</h2>
          <h4 className={classes.aboutFilmTextTagline}>{tagline}</h4>
          <div className={classes.time}>
            <span className={classes.timeItem}>{release_date.slice(0, 4)}</span>
            <span className={classes.timeItem}>{`${runtime} min`}</span>
          </div>
          <p className={classes.overview}>{overview}</p>
        </div>
      </div>
    </div>
  );
};

FilmInfo.propTypes = {
  filmInfo: PropTypes.shape({
    title: PropTypes.string,
    poster_path: PropTypes.string,
    tagline: PropTypes.string,
    release_date: PropTypes.string,
    runtime: PropTypes.number,
    overview: PropTypes.string,
  }).isRequired,
};
