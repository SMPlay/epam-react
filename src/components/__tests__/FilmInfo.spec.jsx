import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { FilmInfo } from "../FilmInfo";

describe("AboutFilm component", () => {
  let props;
  beforeEach(() => {
    props = {
      filmInfo: {
        title: "Tomb Rider",
        poster_path: "",
        tagline: "blablabla",
        release_date: "2020-05-31",
        runtime: 150,
        overview: "blablabla",
      },
    };
  });

  it("should rendrer with initial state", () => {
    const tree = renderer.create(
      <Router>
        <FilmInfo {...props} />
      </Router>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
