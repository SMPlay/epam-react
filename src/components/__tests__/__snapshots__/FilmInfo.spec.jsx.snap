// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutFilm component should rendrer with initial state 1`] = `
<div>
  <div
    className="navigation-0-2-1"
  >
    <div>
      <a
        href="/"
        onClick={[Function]}
      >
        <h1
          className="logo-0-2-10"
        >
          netflixroulette
        </h1>
      </a>
    </div>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="btn-0-2-11 btn-d0-0-2-14 btn_primary-0-2-12 btn_primary-d1-0-2-15 "
        data-test-button="go-home"
        onClick={[Function]}
      >
        search
      </button>
    </a>
  </div>
  <div
    className="aboutFilm-0-2-2"
  >
    <img
      alt="Tomb Riderblablabla"
      className="aboutFilmImg-0-2-3"
      src=""
    />
    <div
      className="aboutFilmText-0-2-4"
    >
      <h2
        className="aboutFilmTextTitle-0-2-5"
      >
        Tomb Rider
      </h2>
      <h4
        className="aboutFilmTextTagline-0-2-6"
      >
        blablabla
      </h4>
      <div
        className="time-0-2-7"
      >
        <span
          className="timeItem-0-2-8"
        >
          2020
        </span>
        <span
          className="timeItem-0-2-8"
        >
          150 min
        </span>
      </div>
      <p
        className="overview-0-2-9"
      >
        blablabla
      </p>
    </div>
  </div>
</div>
`;
